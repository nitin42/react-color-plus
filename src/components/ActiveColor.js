import React from 'react'
import { css } from 'emotion'
import PropTypes from 'prop-types'

import { getContrastingColor } from '../utils/colors'

// Display the hex code for the currently selected color
const ActiveColor = ({ color }) => (
  <div
    className={css`
      font-size: 18px;
      position: relative;
    `}
    style={{
      color: getContrastingColor(color)
    }}
  >
    {color}
  </div>
)

ActiveColor.propTypes = {
  color: PropTypes.string.isRequired
}

export default ActiveColor
