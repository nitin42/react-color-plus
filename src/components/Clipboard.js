import React from 'react'

const Clipboard = ({ copyColor, color }) => (
	<span title="clipboard picker" onClick={copyColor}>
		<i id="image-icon" className="fas fa-clipboard" style={{ marginLeft: 10, color }} />
	</span>
)

export default Clipboard
