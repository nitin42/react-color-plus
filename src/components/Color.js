import React from 'react'
import { css } from 'emotion'

const Color = props => (
	<div
		className={css`
			background: ${props.value};
			height: 22px;
			width: 22px;
			cursor: pointer;
			position: relative;
			outline: none;
			float: left;
			margin-right: 10px;
			margin-bottom: 10px;
			border-radius: 4px;
			&:focus {
				box-shadow: ${props.value} 0px 0px 4px;
			}
		`}
		onClick={props.updateColor}
		tabIndex={0}
	/>
)

// box-shadow: ${props.value} 0px 0px 4px;

export default Color
