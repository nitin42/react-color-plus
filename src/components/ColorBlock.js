import React from 'react'
import styled from 'react-emotion'

const Block = styled('div')`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  height: 110px;
  background: ${props => props.color.toHexString()};
  border-radius: 6px 6px 0px 0px;
`

// Displays a color block with the active color hex code
const ColorBlock = ({ color, children }) => (
  <Block color={color}>{children}</Block>
)

export default ColorBlock
