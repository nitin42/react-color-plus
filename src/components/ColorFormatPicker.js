import React from 'react'
import { css } from 'emotion'

const renderFormats = formats =>
  formats.map((format, key) => {
    return (
      <option value={format} key={key}>
        {format}
      </option>
    )
  })

const ColorFormatPicker = ({ changeFormat, formats }) => (
  <div
    className={css`
      display: flex;
      justify-content: center;
      margin-top: 15px;
    `}
  >
    <select
      className={css`
        border: 0px solid rgb(102, 102, 102);
        box-sizing: border-box;
        padding: 0px 7px;
        border-radius: 4px;
        color: rgb(102, 102, 102);
        height: 18px;
        box-shadow: rgb(221, 221, 221) 0px 0px 0px 1px inset;
        outline: none;
        font-size: 12px;
        color: rgb(102, 102, 102);
      `}
      name="formats"
      onChange={changeFormat}
    >
      {renderFormats(formats)}
    </select>
  </div>
)

export default ColorFormatPicker
