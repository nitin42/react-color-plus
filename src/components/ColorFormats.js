import React from 'react'

const ColorFormats = props => (
	<select style={inputStyles} name="formats" onChange={e => this.setState({ currentFormat: e.target.value })}>
		{this.renderFormats()}
	</select>
)

export default ColorFormats
