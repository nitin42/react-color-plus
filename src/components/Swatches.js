import React from 'react'
import { css } from 'emotion'
import PropTypes from 'prop-types'

import Swatch from './Swatch'

const ENTER_KEY = 13

const Swatches = ({ swatches, updateSwatch, onSwatchHover }) => (
  <div
    className={css`
      display: grid;
      grid-template-columns: repeat(6, 1fr);
      grid-gap: 5.3px;
    `}
  >
    {swatches.map((color, key) => (
      <Swatch
        key={key}
        color={color}
        updateSwatch={e => updateSwatch(color, e)}
        updateSwatchOnKeyDown={e =>
          e.keyCode === ENTER_KEY ? updateSwatch(color, e) : null
        }
        onSwatchHover={e => onSwatchHover && onSwatchHover(color, e)}
      />
    ))}
  </div>
)

Swatches.propTypes = {
  swatches: PropTypes.arrayOf(PropTypes.string),
  updateSwatch: PropTypes.func,
  onSwatchHover: PropTypes.func
}

export default Swatches
