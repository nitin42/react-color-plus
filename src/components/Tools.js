import React from 'react'
import PropTypes from 'prop-types'
import styled from 'react-emotion'

import Slider from './Slider'
import { Consumer } from '../utils/context'
import {
  SaturatorIcon,
  DesaturatorIcon,
  ColorSpinIcon,
  ColorDarkenerIcon,
  ColorBrightenerIcon,
  ImagePickerIcon,
  SwatchesGeneratorIcon,
  ShadesGeneratorIcon,
  ResetIcon,
  TintsGeneratorIcon,
  ClipboardIcon,
  GenerateGradientIcon
} from '../icons/index'

// Copied from primer/primer-tooltips/build
import '../styles/tooltip.css'

const StyledSpan = styled('span')`
  outline: none;
`

const StyledLabel = styled('label')`
  display: inline-block;
  width: 10px;
  position: relative;
  top: 3px;
  left: 4px;
  color: ${props => props.color};
`

const ENTER_KEY = 13

/* eslint-disable operator-linebreak */
const TOOLTIP_CLASSNAME =
  'tooltipped tooltipped-ne tooltipped-align-left-1 tooltipped-no-delay border p-2 mb-2 mr-2 float-left'

const HOC = Comp => props => (
  <Consumer>{color => <Comp color={color} {...props} />}</Consumer>
)

const ColorSaturator = HOC(({ color, value, onChange }) => (
  <StyledSpan title="color saturator">
    <SaturatorIcon width={20} height={20} color={color} />
    <Slider
      min="10"
      max="100"
      value={value}
      onChange={onChange}
      color={color}
    />
    <StyledLabel color={color}>{value}</StyledLabel>
  </StyledSpan>
))

ColorSaturator.displayName = 'ColorSaturator'

const ColorDesaturator = HOC(({ color, value, onChange }) => (
  <StyledSpan title="color desaturator">
    <DesaturatorIcon width={20} height={20} color={color} />
    <Slider
      min="10"
      max="100"
      value={value}
      onChange={onChange}
      color={color}
    />
    <StyledLabel color={color}>{value}</StyledLabel>
  </StyledSpan>
))

ColorDesaturator.displayName = 'ColorDesaturator'

const ColorBrightener = HOC(({ color, value, onChange }) => (
  <StyledSpan title="color brightener">
    <ColorBrightenerIcon width={20} height={20} color={color} />
    <Slider
      min="10"
      max="100"
      value={value}
      onChange={onChange}
      color={color}
    />
    <StyledLabel color={color}>{value}</StyledLabel>
  </StyledSpan>
))

ColorBrightener.displayName = 'ColorBrightener'

const ColorDarkener = HOC(({ color, value, onChange }) => (
  <StyledSpan title="color darkener">
    <ColorDarkenerIcon width={16} height={16} color={color} />
    <Slider
      min="10"
      max="100"
      value={value}
      onChange={onChange}
      color={color}
    />
    <StyledLabel color={color}>{value}</StyledLabel>
  </StyledSpan>
))

ColorDarkener.displayName = 'ColorDarkener'

const ColorSpinner = HOC(({ color, onChange, value }) => (
  <StyledSpan title="color spin">
    <ColorSpinIcon width={18} height={18} color={color} />
    <Slider
      min="-360"
      max="360"
      value={value}
      onChange={onChange}
      color={color}
    />
    <StyledLabel color={color}>
      {value}
      <sup>&deg;</sup>
    </StyledLabel>
  </StyledSpan>
))

ColorSpinner.displayName = 'ColorSpinner'

const ImagePicker = HOC(({ color, uploadImage }) => (
  <StyledSpan title="Image picker">
    <input
      id="uploader"
      style={{ display: 'none' }}
      type="file"
      accept="image/*"
      onChange={uploadImage}
    />
    <ImagePickerIcon id="image-icon" height={20} width={20} color={color} />
  </StyledSpan>
))

ImagePicker.propTypes = {
  color: PropTypes.string,
  uploadImage: PropTypes.func
}

const PaletteGenerator = HOC(({ color, generateSwatches }) => (
  <StyledSpan
    tabIndex={0}
    title="Palette generator"
    onClick={generateSwatches}
    onKeyDown={e => (e.keyCode === ENTER_KEY ? generateSwatches(e) : null)}
  >
    <SwatchesGeneratorIcon width={21} height={21} color={color} />
  </StyledSpan>
))

PaletteGenerator.displayName = 'SwatchesGenerator'

PaletteGenerator.propTypes = {
  color: PropTypes.string,
  generateSwatches: PropTypes.func
}

const ShadesGenerator = HOC(({ color, generateShades }) => (
  <StyledSpan
    tabIndex={0}
    title="shade picker"
    onClick={generateShades}
    onKeyDown={e => (e.keyCode === ENTER_KEY ? generateShades(e) : null)}
  >
    <ShadesGeneratorIcon height={20} width={20} color={color} />
  </StyledSpan>
))

ShadesGenerator.displayName = 'ShadesGenerator'

ShadesGenerator.propTypes = {
  color: PropTypes.string,
  generateShades: PropTypes.func
}

const Reset = HOC(({ color, resetColors }) => (
  <StyledSpan
    tabIndex={0}
    title="reset colors"
    onClick={resetColors}
    onKeyDown={e => (e.keyCode === ENTER_KEY ? resetColors(e) : null)}
  >
    <ResetIcon height={18} width={17.5} color={color} />
  </StyledSpan>
))

Reset.displayName = 'Reset'

Reset.propTypes = {
  color: PropTypes.string,
  resetColors: PropTypes.func
}

const TintsGenerator = HOC(({ color, generateTints }) => (
  <StyledSpan
    tabIndex={0}
    title="tint picker"
    onClick={generateTints}
    onKeyDown={e => (e.keyCode === ENTER_KEY ? generateTints(e) : null)}
  >
    <TintsGeneratorIcon height={21} width={21} color={color} />
  </StyledSpan>
))

TintsGenerator.displayName = 'TintsGenerator'

TintsGenerator.propTypes = {
  color: PropTypes.string,
  generateTints: PropTypes.func
}

const Clipboard = HOC(({ color, copyColor, showMsg, id = 'clipboard' }) => (
  <StyledSpan
    tabIndex={0}
    id={id}
    onClick={copyColor}
    onKeyDown={e => (e.keyCode === ENTER_KEY ? copyColor(e) : null)}
    className={showMsg ? TOOLTIP_CLASSNAME : 'no-tooltip'}
    aria-label="Copied"
  >
    <ClipboardIcon height={18} width={19} color={color} />
  </StyledSpan>
))

Clipboard.propTypes = {
  color: PropTypes.string,
  copyColor: PropTypes.func,
  showMsg: PropTypes.bool,
  id: PropTypes.string
}

const GradientGenerator = HOC(({ color, generateGradient }) => (
  <StyledSpan
    tabIndex={0}
    onClick={generateGradient}
    title="Gradient generator"
    onKeyDown={e => (e.keyCode === ENTER_KEY ? generateGradient(e) : null)}
  >
    <GenerateGradientIcon height={20} width={17.5} color={color} />
  </StyledSpan>
))

GradientGenerator.propTypes = {
  color: PropTypes.string,
  generateGradient: PropTypes.func
}

const AdvanceTools = {
  ColorSaturator,
  ColorDesaturator,
  ColorBrightener,
  ColorDarkener,
  ColorSpinner
}

Object.keys(AdvanceTools).forEach(tool => {
  AdvanceTools[tool].propTypes = {
    color: PropTypes.string,
    value: PropTypes.number,
    onChange: PropTypes.func
  }
})

const BasicTools = {
  TintsGenerator,
  ShadesGenerator,
  Reset,
  ImagePicker,
  PaletteGenerator,
  Clipboard,
  GradientGenerator
}

export { AdvanceTools, BasicTools }
