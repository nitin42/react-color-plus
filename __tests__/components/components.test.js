import React from 'react'
import renderer from 'react-test-renderer'
import { TinyColor } from '@ctrl/tinycolor'

import ColorBlock from '../../src/components/ColorBlock'
import Swatches from '../../src/components/Swatches'
import { AdvanceTools, BasicTools } from '../../src/components/Tools'

import { Provider } from '../../src/utils/context'

const noop = () => {}

it('should render the color block with a color code and children', () => {
  const App = () => (
    <ColorBlock colorState={new TinyColor('#f00')} color={'#f00'} />
  )

  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render the color block with hex color when the current format is HSV', () => {
  const App = () => (
    <ColorBlock
      colorState={new TinyColor('#f00')}
      currentFormat={'HSV'}
      color={'#f00'}
    />
  )

  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render a list of swatches', () => {
  const App = () => (
    <Swatches
      swatches={['red', 'blue', 'purple', 'green', 'brown']}
      updateSwatch={noop}
      onSwatchHover={noop}
    />
  )

  const tree = renderer.create(<App />).toJSON()

  expect(tree).toMatchSnapshot()
})

it('should render basic color tools', () => {
  const color = '#f00'

  const BTools = () => (
    <Provider value={color}>
      <BasicTools.TintsGenerator generateTints={noop} />
      <BasicTools.ShadesGenerator generateShades={noop} />
      <BasicTools.PaletteGenerator generateSwatches={noop} />
      <BasicTools.ImagePicker uploadImage={noop} />
      <BasicTools.Reset resetColors={noop} />
      <BasicTools.Clipboard copyColor={noop} />
    </Provider>
  )

  const tree = renderer.create(<BTools />).toJSON()

  expect(tree).toMatchSnapshot()
})

it('should render advanced color tools', () => {
  const color = '#eeee'

  const ATools = () => (
    <Provider value={color}>
      <AdvanceTools.ColorBrightener value={0} onChange={noop} />
      <AdvanceTools.ColorDarkener value={20} onChange={noop} />
      <AdvanceTools.ColorDesaturator value={50} onChange={noop} />
      <AdvanceTools.ColorLightener value={10} onChange={noop} />
      <AdvanceTools.ColorSaturator value={20} onChange={noop} />
      <AdvanceTools.ColorSpinner value={-120} onChange={noop} />
    </Provider>
  )

  const tree = renderer.create(<ATools />).toJSON()

  expect(tree).toMatchSnapshot()
})
