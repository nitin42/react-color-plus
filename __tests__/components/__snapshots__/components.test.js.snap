// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of swatches 1`] = `
<div
  className="css-w1azp9"
>
  <div
    className="css-14g5nes"
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="swatch"
    tabIndex={0}
    title="red"
  />
  <div
    className="css-pa4qu5"
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="swatch"
    tabIndex={0}
    title="blue"
  />
  <div
    className="css-8tbn6u"
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="swatch"
    tabIndex={0}
    title="purple"
  />
  <div
    className="css-1o54mkk"
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="swatch"
    tabIndex={0}
    title="green"
  />
  <div
    className="css-18dp5ae"
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="swatch"
    tabIndex={0}
    title="brown"
  />
</div>
`;

exports[`should render advanced color tools 1`] = `
Array [
  <span
    className="css-1v5ga19"
    title="color brightener"
  >
    <i
      className="fas fa-sun css-1makrmg"
      color="#eeee"
    />
    <input
      className="css-qvm90c"
      max="100"
      min="10"
      onChange={[Function]}
      type="range"
      value={0}
    />
    <label
      className="css-142tcgl"
      color="#eeee"
    >
      0
    </label>
  </span>,
  <span
    className="css-1v5ga19"
    title="color darkener"
  >
    <i
      className="fas fa-moon css-1makrmg"
      color="#eeee"
    />
    <input
      className="css-qvm90c"
      max="100"
      min="10"
      onChange={[Function]}
      type="range"
      value={20}
    />
    <label
      className="css-142tcgl"
      color="#eeee"
    >
      20
    </label>
  </span>,
  <span
    className="css-1v5ga19"
    title="color desaturator"
  >
    <i
      className="fas fa-fill css-1makrmg"
      color="#eeee"
    />
    <input
      className="css-qvm90c"
      max="100"
      min="10"
      onChange={[Function]}
      type="range"
      value={50}
    />
    <label
      className="css-142tcgl"
      color="#eeee"
    >
      50
    </label>
  </span>,
  <span
    className="css-1v5ga19"
    title="color saturator"
  >
    <i
      className="fas fa-star-half-alt css-1makrmg"
      color="#eeee"
    />
    <input
      className="css-qvm90c"
      max="100"
      min="10"
      onChange={[Function]}
      type="range"
      value={20}
    />
    <label
      className="css-142tcgl"
      color="#eeee"
    >
      20
    </label>
  </span>,
  <span
    className="css-1v5ga19"
    title="color spin"
  >
    <i
      className="fas fa-sync-alt css-1makrmg"
      color="#eeee"
    />
    <input
      className="css-qvm90c"
      max="360"
      min="-360"
      onChange={[Function]}
      type="range"
      value={-120}
    />
    <label
      className="css-142tcgl"
      color="#eeee"
    >
      -120
      <sup>
        Â°
      </sup>
    </label>
  </span>,
]
`;

exports[`should render basic color tools 1`] = `
Array [
  <span
    className="css-1v5ga19"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    title="tint picker"
  >
    <i
      className="fas fa-tint css-iprkew"
      color="#f00"
      id="image-icon"
    />
  </span>,
  <span
    className="css-1v5ga19"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    title="shade picker"
  >
    <i
      className="fas fa-adjust css-iprkew"
      color="#f00"
      id="image-icon"
    />
  </span>,
  <span
    className="css-1v5ga19"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    title="Palette generator"
  >
    <i
      className="fas fa-palette css-iprkew"
      color="#f00"
      id="image-icon"
    />
  </span>,
  <span
    className="css-1v5ga19"
    title="Image picker"
  >
    <input
      accept="image/*"
      id="uploader"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <i
      className="fas fa-image css-iprkew"
      color="#f00"
      id="image-icon"
    />
  </span>,
  <span
    className="css-1v5ga19"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    title="reset colors"
  >
    <i
      className="fas fa-arrow-alt-circle-left css-iprkew"
      color="#f00"
      id="image-icon"
    />
  </span>,
  <span
    aria-label="Copied"
    className="no-tooltip css-1v5ga19"
    id="clipboard"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <i
      className="fas fa-copy css-iprkew"
      color="#f00"
      id="image-icon"
    />
  </span>,
]
`;

exports[`should render the color block with a color code and children 1`] = `
<div
  className="css-3ef40x"
  style={
    Object {
      "background": "#f00",
    }
  }
>
  <div
    className="css-1uaa0vj"
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    #f00
  </div>
</div>
`;

exports[`should render the color block with hex color when the current format is HSV 1`] = `
<div
  className="css-3ef40x"
  style={
    Object {
      "background": "#ff0000",
    }
  }
>
  <div
    className="css-1uaa0vj"
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    #f00
  </div>
</div>
`;
